pgmpy/models/FactorGraph.py

```diff
@@ -134,11 +137,69 @@ def remove_factors(self, *factors):
         >>> G = FactorGraph()
         >>> G.add_nodes_from(['a', 'b', 'c'])
         >>> phi1 = DiscreteFactor(['a', 'b'], [2, 2], np.random.rand(4))
+        >>> G.add_edge('a',phi1)
         >>> G.add_factors(phi1)
         >>> G.remove_factors(phi1)
         """
         for factor in factors:
             self.factors.remove(factor)
+            if factor in super(FactorGraph, self).nodes():
+                super(FactorGraph, self).remove_node(factor)
```

Comment:
I think this needs to be done with every remove_factor method in every model you can look at #752  for explanation